target/linux/mpc85xx/files/arch/powerpc/boot/dts/ecw7210.dts

```diff
@@ -0,0 +1,304 @@
+ * Free Software Foundation;  either version 2 of the  License, or (at your
+ * option) any later version.
+ */
+
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/include/ "fsl/p1020si-pre.dtsi"
+/ {
+	model = "Edge-Core ECW7210-L";
+	compatible = "edgecore,ecw7210";
+
+	memory {
+		device_type = "memory";
+	};
+
+	board_lbc: lbc: localbus@ffe05000 {
+		reg = <0 0xffe05000 0 0x1000>;
+		ranges = <0x0 0x0 0x0 0xee000000 0x2000000
+			  0x1 0x0 0x0 0xffa00000 0x0040000>;
+		interrupts = <0x13 0x2>;
+		#size-cells = <0x1>;
+		#address-cells = <0x2>;
+		bus-frequency = <24999999>;
+		nand@1,0 {
+			#address-cells = <0x1>;
+			#size-cells = <0x1>;
+			compatible = "fsl,p1020-fcm-nand", "fsl,elbc-fcm-nand";
+			reg = <0x1 0x0 0x40000>;
+			partition@0 {
+				reg = <0x0 0x1000000>;
+				label = "kernel";
+			};
+			partition@01000000 {
+				reg = <0x1000000 0x1f000000>;
+				label = "ubi";
+			};
+			
+			//label = "ubi";
+			//compatible = "ubi,volume";
+		};
+		
+		// may be best to leave the NOR flash alone, since we can put everything in NAND.
+		nor@0,0 {
+			device-width = <0x1>;
+			bank-width = <0x2>;
+			reg = <0x0 0x0 0x2000000>;
+			compatible = "cfi-flash";
+			#size-cells = <0x1>;
+			#address-cells = <0x1>;
+
+			partition@1f80000 {
+				read-only;
+				label = "mtd-uboot-image";
+				reg = <0x1f80000 0x80000>;
+			};
+
+			partition@1f60000 {
+				label = "uboot-env";
+				reg = <0x1f60000 0x20000>;
+			};
+
+			partition@1f40000 {
+				read-only;
+				label = "u-env-backup";
+				reg = <0x1f40000 0x20000>;
+			};
+
+			partition@1f20000 {
+				read-only;
+				label = "mtd-mfr";
+				reg = <0x1f20000 0x20000>;
+			};
+
+			partition@1e20000 {
+				read-only;
+				label = "mtd-reserved";
+				reg = <0x1e20000 0x100000>;
+			};
+
+			partition@1e00000 {
+				read-only;
+				label = "crashlog";
+				reg = <0x1e00000 0x20000>;
+			};
+
+			partition@60000 {
+				read-only;
+				label = "mtd-unused";
+				reg = <0x60000 0x1da0000>;
+			};
+
+			partition@20000 {
+				read-only;
+				label = "mtd-dtb";
+				reg = <0x20000 0x40000>;
+			};
+
+			partition@0 {
+				read-only;
+				label = "mtd-u-script";
+				reg = <0x0 0x20000>;
+			};
+		};
+	};
+
+	board_soc: soc: soc@ffe00000 {
+		ranges = <0x0 0x0 0xffe00000 0x100000>;
+
+		i2c@3100 {
+			status = "okay";
+		};
+
+                mdio@25000 {
+			compatible = "fsl,etsec2-tbi";
+			tbi_phy1: tbi-phy@12 {
+				reg = <0x12>;
+			};	
+		};
+
+		
+		// We seem to need to have the tbi_phy's laid out this way for enet2 to work
+		mdio@24000 {
+			compatible = "fsl,etsec2-mdio";
+
+			phy0: ethernet-phy@0 {
+				reg = <0x0>;
+			};
+
+			phy1: ethernet-phy@1 {
+				reg = <0x1>;
+			};
+
+			tbi_phy0: tbi-phy@11 {
+				reg = <0x11>;
+			};
+		};
+
+
+		
+                eth0: enet2: ethernet@b2000 {
+			status = "okay";
+			phy-connection-type = "rgmii-id";
+			model = "eTSEC";
+			phy-handle = <&phy1>;
+			tbi-handle = <&tbi_phy1>;
+			local-mac-address = [70 72 cf 42 42 03];		
```

Comment:
You are setting a static MAC address here??

Reply:
It looks like I need some kind of MAC here, even if it's made up. U-boot will replace it if everything works correctly (if ethaddr and ethaddr2 are set). I had considered 00:00:00:00:00:00, but if u-boot doesn't set it, we end up without a MAC on eth0, and that causes br-lan not to come up. 

70:72:CF is Edge-core's OUI, and I figured 42:42:xx would be as good of a MAC as any.

I figured that we'd be better off with a made-up MAC than an inaccessible device, but anyone working on this will undoubtedly have a console. I'll defer to your judgement - just let me know if you want these two MACs taken out.

Reply:
You have to _read_ the MAC address from some location. Have look at how this is done for other devices. I've made an incomplete tutorial here: https://openwrt.org/docs/guide-developer/mac.address

Reply:
In the stock system, the MAC addresses appear to come from u-boot environment variables. It's not actively read by the OS from anywhere, as on other systems - it appears to be copied into the stock DTS by u-boot, on top of whatever value is entered into those `local-mac-address` properties.

Now, I could be wrong. Changing those u-boot environment variables seemed to change the MAC's in the DTS got handed to the kernel, but it's also possible that those MACs can be found somewhere in the NOR or NAND flash. I haven't done that search, and need to. I can also confirm that `ethaddr` and `ethaddr2` are set with the correct MAC's on a stock system, and that they're not junk left in the environment from the stock setup.

I'll confirm that I have to have the `local-mac-address` property for the u-boot-DTS-MAC-copying to work. I can just set it to all 00's, which effectively disables the interface if it u-boot doesn't set it. (the eTSEC driver appears to disallow `ifconfig up` on the interface if there's no MAC address). Would that be the best choice if the property has to be there?

Reply:
The MAC's are no longer set in the dts. This leaves a non-functional router if the u-boot environment gets reset for some reason. The stock OS really does rely on the MAC addresses being set by u-boot, but they work around u-boot environment issues by aggressively detecting and re initializing the u-boot environment if it's changed (including re-locking u-boot's autoboot!).

One note: the stock OS uses `ethaddr` and eth1addr` to set MACs. Under Openwrt, we need to instead use `ethaddr` and `eth2addr` to set the MACs. I suspect this is because of the way u-boot is searching through the device tree. I've clarified info about setting those environment variables to the commit message.